apps/backend/src/integrations.rs

```diff
@@ -52,6 +52,124 @@ impl IntegrationService {
+            use super::*;
+
+            #[derive(Serialize, Deserialize, Debug, Clone)]
+            #[serde(rename_all = "PascalCase")]
+            pub struct EmbyWebhookPlaybackInfoPayload {
+                pub position_ticks: Option<Decimal>,
+            }
+            #[derive(Serialize, Deserialize, Debug, Clone)]
+            #[serde(rename_all = "PascalCase")]
+            pub struct EmbyWebhookItemProviderIdsPayload {
+                pub tmdb: Option<String>,
+            }
+            #[derive(Serialize, Deserialize, Debug, Clone)]
+            #[serde(rename_all = "PascalCase")]
+            pub struct EmbyWebhookItemPayload {
+                pub run_time_ticks: Option<Decimal>,
+                #[serde(rename = "Type")]
+                pub item_type: String,
+                pub provider_ids: EmbyWebhookItemProviderIdsPayload,
+                #[serde(rename = "ParentIndexNumber")]
+                pub season_number: Option<i32>,
+                #[serde(rename = "IndexNumber")]
+                pub episode_number: Option<i32>,
+                #[serde(rename = "Name")]
+                pub episode_name:  Option<String>,
+                pub series_name: Option<String>,
+            }
+            #[derive(Serialize, Deserialize, Debug, Clone)]
+            #[serde(rename_all = "PascalCase")]
+            pub struct EmbyWebhookPayload {
+                pub event: Option<String>,
+                pub item: EmbyWebhookItemPayload,
+                pub series: Option<EmbyWebhookItemPayload>,
+                pub playback_info: EmbyWebhookPlaybackInfoPayload,
+            }
+        }
+
+        let payload = serde_json::from_str::<models::EmbyWebhookPayload>(payload)?;
+
+        let identifier = if let Some(id) = payload.item.provider_ids.tmdb.as_ref() {
+            Some(id.clone())
+        } else {
+            payload
+                .series
+                .as_ref()
+                .and_then(|s| s.provider_ids.tmdb.clone())
+        };
+
+        if payload.item.run_time_ticks.is_none() {
+            bail!("No run time associated with this media")
+        }
+        if payload.playback_info.position_ticks.is_none() {
+            bail!("No position associated with this media")
+        }
+
+        let runtime = payload.item.run_time_ticks.unwrap();
+        let position = payload.playback_info.position_ticks.unwrap();
+
+        let (identifier, lot) = match payload.item.item_type.as_str() {
+            "Episode" => {
+                if payload.item.episode_name.is_none() {
+                    bail!("No episode name associated with this media")
+                }
+
+                if payload.item.series_name.is_none() {
+                    bail!("No series name associated with this media")
+                }
+
+                let series_name = payload.item.series_name.unwrap();
+                let episode_name = payload.item.episode_name.unwrap();
+
+                let db_show = Metadata::find()
```

Comment:
Since this is repeated, please move them into one common method in the service.

Reply:
They are slightly difference so I moved it into a function taking a conditional parameter.